.github/workflows/README.md

```diff
@@ -187,3 +205,254 @@ jobs:
+
+### Prerequisites
+
+Your Java project needs to be set up with Gradle and either needs to contain a `build.gradle` or a `build.gradle.kts`
+file that uses the `plugin-publish-plugin` dependency. Moreover, prepare credentials for Sonarcloud, Sonatype, GitHub
+and Gradle
+Plugin Portal.
+
+### Dependencies
+
+This workflow is built from multiple composite actions listed below:
+
+- [java-gradle-build](https://github.com/bakdata/ci-templates/tree/main/actions/java-gradle-build)
+- [java-gradle-test](https://github.com/bakdata/ci-templates/tree/main/actions/java-gradle-test)
+- [java-gradle-publish](https://github.com/bakdata/ci-templates/tree/main/actions/java-gradle-publish)
+- [java-gradle-release-github](https://github.com/bakdata/ci-templates/tree/main/actions/java-gradle-release-github)
+
+### Input Parameters
+
+| Name              | Required | Default Value |  Type  | Description                                |
+| ----------------- | :------: | :-----------: | :----: | ------------------------------------------ |
+| java-distribution |    ❌    |   microsoft   | string | Java distribution to be installed          |
+| java-version      |    ❌    |      11       | string | Java version to be installed               |
+| gradle-version    |    ❌    |    wrapper    | string | Gradle version to be installed             |
+| working-directory |    ❌    |      ./       | string | Working directory of your Gradle artifacts |
+
+### Secret Parameters
+
+For Sonarcloud you need to provide a `sonar-token` to publish code quality results. In case of Sonatype, the action
+requires you to have a `signing-secret-key-ring` (base64 encoded), a `signing-key-id` and a `signing-password` to sign
+your build artifacts and additionally a `ossrh-username` and a `ossrh-password` to publish the signed artifacts to
+Nexus. To publish the Gradle plugin to the Gradle Plugin Portal you need to provide a `gradle-publish-key` and
+a `gradle-publish-secret`. The `github-username` and `github-token` is required to query the GitHub API for generating a
+changelog when running on a tag branch.
+
+| Name                    | Required | Description                                                    |
+| ----------------------- | :------: | -------------------------------------------------------------- |
+| sonar-token             |    ✅    | Token for Sonarcloud                                           |
+| signing-secret-key-ring |    ✅    | Key ring (base64 encoded) for signing the Sonatype publication |
+| signing-key-id          |    ✅    | Key id for signing the Sonatype publication                    |
+| signing-password        |    ✅    | Password for signing the Sonatype publication                  |
+| ossrh-username          |    ✅    | Username for signing into Sonatype repository                  |
+| ossrh-password          |    ✅    | Password for signing into Sonatype repository                  |
```

Comment:
Please point out that the empty value will skip all the Sonar related jobs! Otherwise, I would think that I need Sonarcloud!